cleverhans/experimental/certification/optimization.py

```diff
@@ -181,31 +264,39 @@ def run_one_step(self, eig_init_vec_val, eig_num_iter_val,
       if LOWER_CERT_BOUND < current_certificate < 0:
         tf.logging.info('Found certificate of robustness!')
         return True
-     # Running step
-    step_feed_dict = {self.eig_init_vec_placeholder: eig_init_vec_val,
-                      self.eig_num_iter_placeholder: eig_num_iter_val,
-                      self.smooth_placeholder: smooth_val,
-                      self.penalty_placeholder: penalty_val}
-
-    [_, self.current_total_objective, self.current_unconstrained_objective,
-     self.current_eig_vec_val,
-     self.current_eig_val_estimate] = self.sess.run(
-         [self.opt_one_step, self.total_objective,
-          self.dual_object.unconstrained_objective,
-          self.eig_vec_estimate,
-          self.eig_val_estimate], feed_dict=step_feed_dict)
+    # Running step
+    step_feed_dict = {
+        self.eig_init_vec_placeholder: eig_init_vec_val,
+        self.eig_num_iter_placeholder: eig_num_iter_val,
+        self.smooth_placeholder: smooth_val,
+        self.penalty_placeholder: penalty_val
+    }
+
+    [
+        _, self.current_total_objective, self.current_unconstrained_objective,
+        self.current_eig_vec_val, self.current_eig_val_estimate
+    ] = self.sess.run([
+        self.opt_one_step, self.total_objective,
+        self.dual_object.unconstrained_objective, self.eig_vec_estimate,
+        self.eig_val_estimate
+    ],
+                      feed_dict=step_feed_dict)
 
     if self.current_step % self.params['print_stats_steps'] == 0:
-      stats = {'total_objective': float(self.current_total_objective),
-               'unconstrained_objective':
-                   float(self.current_unconstrained_objective),
-               'min_eig_val_estimate': float(self.current_eig_val_estimate)}
+      stats = {
+          'total_objective':
+              float(self.current_total_objective),
+          'unconstrained_objective':
+              float(self.current_unconstrained_objective),
+          'min_eig_val_estimate':
+              float(self.current_eig_val_estimate)
+      }
       tf.logging.debug('Current inner step: %d, optimization stats: %s',
                        self.current_step, stats)
       if self.params['stats_folder'] is not None:
         stats = json.dumps(stats)
-        with tf.gfile.Open((self.params['stats_folder']+ '/' +
-                            str(self.current_step) + '.json'))as file_f:
+        with tf.gfile.Open((self.params['stats_folder'] + '/' + str(
```

Comment:
to fit `with tf.gfile.Open(...)` into single line you can introduce `filename` variable and then just use `with tf.gfile.Open(filename, 'w'):`

also please use `os.path.join(a, b)` instead of `a + '/' + b`